### ISSUE TITLE
[Feature Request]: Windows BLE Support

### ISSUE THREAD

**Issue by aritchie** [shinyorg/shiny#1232]
### Summary

None

### API Changes

Some methods may have to become async to deal with windows API

### Intended Use Case

Windows support

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by beeradmoore** [comment]
Hey @aritchie ,
This issue was created a few days after the last set of changes for [Shiny.BluetoothLE for Windows](https://github.com/shinyorg/shiny/blob/master/src/Shiny.BluetoothLE/Platforms/Windows/BleManager.cs). Do you recall why `Scan` and other such methods were changed to throw `NotImplementedException`?

Regarding what you wrote here,
> Some methods may have to become async to deal with windows API

Has this changed since Shiny v3 release or is the methods you are talking about the internal Windows ones?

**Comment by aritchie** [comment]
> This issue was created a few days after the last set of changes for [Shiny.BluetoothLE for Windows](https://github.com/shinyorg/shiny/blob/master/src/Shiny.BluetoothLE/Platforms/Windows/BleManager.cs). Do you recall why Scan and other such methods were changed to throw NotImplementedException?

Yes because I haven't evaluated all of the API changes yet.  I'm not just copying things one-to-one over.

> Has this changed since Shiny v3 release or is the methods you are talking about the internal Windows ones?

I'm not sure yet.  This isn't finished.

**Comment by beeradmoore** [comment]
All good. Thanks for the update on where it's and and why it's at that state.

**Comment by bcaceiro** [comment]
Hello @aritchie ! Though StopScanning is not implemented in windows, should the  bleManager.Scan find any devices on Windows? I have this working on iOS, and Android (MAUI Project), but on Windows it doesn't find any device. going through the docs I can't find any specific setting, am I missing something or it is not supposed to work? Best regards and thanks for the great job !

**Comment by aritchie** [comment]
@bcaceiro The docs don't say anything because windows isn't supported anywhere in Shiny yet.  That is the whole purpose of this feature request.  If you want to know the specific status, you can follow the source code or wait until I mark this ticket

**Comment by gcadmes-extron** [comment]
@bcaceiro  if your looking for just discovery on windows, you might want to implement your own wrapper around Shiny. I had to do that as I couldn't wait for Shiny to implement support for Windows. 

Because I only needed advertisements, I ended up  using 'BluetoothLEAdvertisementWatcher' 
Here is a [simple example](https://riptutorial.com/bluetooth/example/19735/listen-for-a-bluetooth-le-advertisement) for only advertisements.

**Comment by BillyMartin1964** [comment]
Any update on whether windows is supported now?

**Comment by aritchie** [comment]
Please refer to the source code & nuget packages for updates.

Currently, I only keep Windows support for my customers.  I may open source it at a later date.

**Comment by aritchie** [comment]
There is now Windows support in v4
